packages/text-field/index.tsx

```
@@ -209,41 +213,33 @@ class TextField<T extends HTMLElement = HTMLInputElement> extends React.Componen
     );
   }
 
-  get inputAdapter(): Partial<MDCTextFieldAdapter> {
-    // For reference: This is the shape of what the vanilla component `getNativeInput` returns
-    // {
-    //  value: string,
-    //  disabled: boolean, --> doesn't need to be implemented since the <INPUT> handles it
-    //  also the `get disabled` isn't actually used, except in the vanilla component
-    //  validity: {
-    //    badInput: boolean,
-    //    valid: boolean,
-    //  },
-    // }
+  get inputAdapter(): MDCTextFieldInputAdapter {
     return {
+      isFocused: () => this.state.isFocused,
       getNativeInput: () => {
-        let badInput;
-        let valid;
-        if (this.inputComponent_) {
-          badInput = this.inputComponent_.isBadInput();
-          valid = this.inputComponent_.isValid();
+        const component = this.inputComponent_;
+        if (!component) {
+          return null;
```

i think we should actually throw an error here.